src/mds/MDCache.cc

```diff
@@ -11626,11 +11632,28 @@ void MDCache::dump_cache(const char *fn, Formatter *f)
       return;
     }
   }
-  
+
   for (ceph::unordered_map<vinodeno_t,CInode*>::iterator it = inode_map.begin();
        it != inode_map.end();
        ++it) {
     CInode *in = it->second;
+
+    if (!dump_root.empty()) {
+      string ipath;
+      if (in->is_root())
+	ipath = "/";
+      else
+	in->make_path_string(ipath);
```

Comment:
Hmm, this is going to ignore hard links under the requested path filter, but that's probably okay given that we're dumping a flat list of inodes anyway, and not pretending that this is a tree-like output.